src/NuGet.Packaging.Core/PackageReaderCoreBase.cs

```diff
@@ -31,6 +31,8 @@ public virtual NuGetVersion GetMinClientVersion()
             return NuspecCore.GetMinClientVersion();
         }
 
+        public virtual PackageType GetPackageType() => NuspecCore.GetPackageType();
+
         public abstract Stream GetStream(string path);
```

Comment:
while you are here, move the abstract guys up and kill the unnecessary constructor